src/catalua.cpp

```diff
@@ -1211,14 +1211,20 @@ void use_function::dump_info( const item &it, std::vector<iteminfo> &dump ) cons
     }
 }
 
-long use_function::call( player &p, item &it, bool active, const tripoint &pos ) const
+ret_val<bool> use_function::can_call(const player &p, const item &it, bool t, const tripoint &pos) const
 {
     if( actor == nullptr ) {
         if( p.is_player() ) {
-            add_msg(_("You can't do anything interesting with your %s."), it.tname().c_str());
+            ret_val<bool>::make_failure( _( "You can't do anything interesting with your %s." ), it.tname().c_str() );
         }
-        return 0;
+        return ret_val<bool>::make_failure();
```

Comment:
This separate return value for non-players can probably be removed (it was needed to avoid the `add_msg` call, but that is now done and checked in the caller). The code in the `iuse_actor::can_use` functions returns the same same (non-empty) error string for npcs anyway.